docs/usage.md

```diff
@@ -1 +1,50 @@
 # Usage
+
+Except for the [legacy API](api/legacy.md), [Apache Arrow](https://arrow.apache.org/) is used as the in-memory interchange format between all formats. While some end-to-end helper functions are provided, the user can go through Arrow objects for maximal flexibility in the conversion process.
+
+All functionality that goes through Arrow is currently exported via the `stac_geoparquet.arrow` namespace.
```

Comment:
I think that an example here (or on the index page?) from STAC -> geoparquet would be valuable.

We have the one currently at https://github.com/stac-utils/stac-geoparquet?tab=readme-ov-file#usage. Even an example using a simple `pystac.Item()` built locally would be good to have.

Reply:
I added a simple example below